doc/design/parallel-node-config.md

```diff
@@ -0,0 +1,207 @@
+5. on Openshift platform we will pause the machine config pool related to the node
+6. the operator will start the drain process
+   1. if `Drain_Required` the operator will remove ONLY pods used sriov devices
+   2. if `Reboot_Required` the operator will remove ALL the pods on the system
+9. operator moves the `sriovnetwork.openshift.io/current-state` annotation to `DrainComplete`
+10. daemon will continue to the configuration when it's done it will move back both `sriovnetwork.openshift.io/state` 
+annotation on Node and `sriovnetwork.openshift.io/desired-state` on SriovNetworkNodeState to `Idle`
+11. operator runs the complete drain to remove the cordon and mark the `sriovnetwork.openshift.io/current-state` annotation to `Idle`
+
+### API Extensions
+
+#### Extend existing CR SriovNetworkPoolConfig
+SriovNetworkPoolConfig is used only for OpenShift to provide configuration for
+OVS Hardware Offloading. We can extend it to add configuration for the drain
+pool. E.g.:
+
+```golang
+// SriovNetworkPoolConfigSpec defines the desired state of SriovNetworkPoolConfig
+type SriovNetworkPoolConfigSpec struct {
+    ...
+	
+    // nodeSelector specifies a label selector for Nodes
+    NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+
+    // maxUnavailable defines either an integer number or percentage
+    // of nodes in the pool that can go Unavailable during an update.
+    //
+    // A value larger than 1 will mean multiple nodes going unavailable during
+    // the update, which may affect your workload stress on the remaining nodes.
+    // Drain will respect Pod Disruption Budgets (PDBs) such as etcd quorum guards,
+    // even if maxUnavailable is greater than one.
+    MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+}
+```
+
+```yaml
+apiVersion: v1
+kind: SriovNetworkPoolConfig
+metadata:
+  name: pool-1
+  namespace: network-operator
+spec:
+  maxUnavailable: "20%"
+  nodeSelector:
+    - matchExpressions:
+      - key: some-label
+        operator: In
+        values:
+          - val-2
+    - matchExpressions:
+      - key: other-label
+        operator: "Exists"
+```
+
+Once this change will be implemented `SriovNetworkPoolConfig` configuration will be applied both to vanilla Kubernetes
+and OpenShift clusters.
+
+### Implementation Constraints
+
+Node can only be part of one pool. if the node is not part of any node it will be allocated
+to a virtual default pool with `maxUnavailable` of 1
+
+### Upgrade & Downgrade considerations
+After operator upgrade we have to support `sriovnetwork.openshift.io/state` node annotation and `sriovnetwork.openshift.io/desired-state`
+annotation in the `sriovNetworkNodeState`. in the future we are going to migrate to only using the annotation in the `sriovNetworkNodeState`
+
+There is no change in upgrade from the user point of view as we are going to create a default pool with `maxUnavailable` 1
```

Comment:
i Thought we arent going to create this default obj.

Reply:
yep yep no default pool sorry I miss that line :)

done